src/FxCop/System.Security.Cryptography.Hashing.Algorithms.Analyzers/Core/Security/DoNotUseInsecureCryptographicAlgorithms.cs

```diff
@@ -0,0 +1,118 @@
+        private static readonly LocalizableString s_localizableDoNotUseSHA1Title = DiagnosticHelpers.GetLocalizableResourceString(nameof(SystemSecurityCryptographyHashingAlgorithmsAnalyzersResources.DoNotUseSHA1));
+        private static readonly LocalizableString s_localizableDoNotUseSHA1Description = DiagnosticHelpers.GetLocalizableResourceString(nameof(SystemSecurityCryptographyHashingAlgorithmsAnalyzersResources.DoNotUseSHA1Description));
+        
+
+        internal static DiagnosticDescriptor DoNotUseMD5Rule = CreateDiagnosticDescriptor(DoNotUseMD5RuleId,
+                                                                                          s_localizableDoNotUseMD5Title,
+                                                                                          s_localizableDoNotUseMD5Description);
+        
+        internal static DiagnosticDescriptor DoNotUseSHA1Rule = CreateDiagnosticDescriptor(DoNotUseSHA1RuleId,
+                                                                                           s_localizableDoNotUseSHA1Title,
+                                                                                           s_localizableDoNotUseSHA1Description);
+         
+        protected abstract Analyzer GetAnalyzer(CompilationStartAnalysisContext context, CompilationSecurityTypes cryptTypes);
+
+        private static readonly ImmutableArray<DiagnosticDescriptor> s_supportedDiagnostics = ImmutableArray.Create(DoNotUseMD5Rule, 
+                                                                                                                    DoNotUseSHA1Rule);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics
+            => s_supportedDiagnostics;
+
+        private static DiagnosticDescriptor CreateDiagnosticDescriptor(string ruleId, LocalizableString title, LocalizableString description, string uri = null)
+        {
+            return new DiagnosticDescriptor(ruleId,
+                                            title,
+                                            title,
+                                            DiagnosticCategory.Security,
+                                            DiagnosticSeverity.Warning,
+                                            true,
+                                            description: description,
+                                            helpLinkUri: uri,
+                                            customTags: WellKnownDiagnosticTags.Telemetry);
+        }
+
+        public override void Initialize(AnalysisContext analysisContext)
+        {
+            analysisContext.RegisterCompilationStartAction(
+                (context) =>
+                {
+                    var cryptTypes = new CompilationSecurityTypes(context.Compilation);
+                    if (ReferencesAnyTargetType(cryptTypes))
+                    {
+                        Analyzer analyzer = GetAnalyzer(context, cryptTypes);
```

Comment:
I will remove the receiving variable of that.

Searched and find in some cases, the analyzer will be used in this statement, in our CA3077 rule, which will create a request later:

Analyzer analyzer = GetAnalyzer(context, xmlTypes, version);
context.RegisterSymbolAction(analyzer.AnalyzeSymbol, SymbolKind.NamedType);

This statement appears to be in both CSharp and VB cases, so I guess it get moved into the base class of CA3077.